src/node_worker.cc

```diff
@@ -511,28 +527,23 @@ void Worker::StopThread(const FunctionCallbackInfo<Value>& args) {
 void Worker::Ref(const FunctionCallbackInfo<Value>& args) {
   Worker* w;
   ASSIGN_OR_RETURN_UNWRAP(&w, args.This());
-  if (w->thread_exit_async_)
-    uv_ref(reinterpret_cast<uv_handle_t*>(w->thread_exit_async_.get()));
+  if (w->response_) uv_ref(reinterpret_cast<uv_handle_t*>(w->response_));
 }
 
 void Worker::Unref(const FunctionCallbackInfo<Value>& args) {
   Worker* w;
   ASSIGN_OR_RETURN_UNWRAP(&w, args.This());
-  if (w->thread_exit_async_)
-    uv_unref(reinterpret_cast<uv_handle_t*>(w->thread_exit_async_.get()));
+  if (w->response_) uv_unref(reinterpret_cast<uv_handle_t*>(w->response_));
```

Comment:
Just noticing: These casts aren’t valid :)

Reply:
thanks, fixed this.